### PR TITLE
support unclip border bbox regression

### PR DESCRIPTION
**Pull request by OceanPang**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -104,6 +104,7 @@ venv.bak/
 .mypy_cache/
 
 data/
+data
```
Comment (ZwwWayne) [inline .gitignore:107]: Is this really necessary?

File: mmdet/core/bbox/samplers/random_sampler.py [hunk 2]
```diff
@@ -43,7 +43,6 @@ def random_choice(self, gallery, num):
             Tensor or ndarray: sampled indices.
         """
         assert len(gallery) >= num
-
```
Comment (ZwwWayne) [inline mmdet/core/bbox/samplers/random_sampler.py:46]: May clean the unnecessary modification.

File: mmdet/datasets/pipelines/transforms.py [hunk 3]
```diff
@@ -657,7 +670,9 @@ def __call__(self, results):
         return results
 
     def __repr__(self):
-        return self.__class__.__name__ + f'(crop_size={self.crop_size})'
+        repr_str = self.__class__.__name__ + f'(crop_size={self.crop_size})'
```
Comment (ZwwWayne) [inline mmdet/datasets/pipelines/transforms.py:673]:
```python
f'(crop_size={self.crop_size})' -> f'(crop_size={self.crop_size}, '
```

File: mmdet/datasets/pipelines/transforms.py [hunk 4]
```diff
@@ -1019,6 +1041,7 @@ def __repr__(self):
         repr_str = self.__class__.__name__
         repr_str += f'(min_ious={self.min_ious}, '
         repr_str += f'min_crop_size={self.min_crop_size})'
```
Comment (ZwwWayne) [inline mmdet/datasets/pipelines/transforms.py:1043]:
```python
f'min_crop_size={self.min_crop_size})' -> f'min_crop_size={self.min_crop_size}, '

```

File: mmdet/datasets/pipelines/transforms.py [hunk 5]
```diff
@@ -1597,6 +1625,7 @@ def __repr__(self):
         repr_str += f'to_rgb={self.to_rgb}, '
         repr_str += f'test_mode={self.test_mode}, '
         repr_str += f'test_pad_mode={self.test_pad_mode})'
```
Comment (ZwwWayne) [inline mmdet/datasets/pipelines/transforms.py:1627]: See comments above.

File: mmdet/core/bbox/coder/bucketing_bbox_coder.py [hunk 6]
```diff
@@ -26,20 +26,24 @@ class BucketingBBoxCoder(BaseBBoxCoder):
              To avoid too large offset displacements. Defaults to 1.0.
         cls_ignore_neighbor (bool): Ignore second nearest bucket or Not.
              Defaults to True.
+        clip_border (bool): Whether clip the objects outside the border of
```
Comment (ZwwWayne) [inline mmdet/core/bbox/coder/bucketing_bbox_coder.py:29]: May update the docstring of `clip_border` as below and also change other places.
```python
clip_border (bool, optional): xxxxxx. Defaults to True.
```

File: mmdet/core/bbox/coder/tblr_bbox_coder.py [hunk 7]
```diff
@@ -17,11 +17,14 @@ class TBLRBBoxCoder(BaseBBoxCoder):
           divided with when coding the coordinates. If it is a list, it should
           have length of 4 indicating normalization factor in tblr dims.
           Otherwise it is a unified float factor for all dims. Default: 4.0
+        clip_border (bool): Whether clip the objects outside the border of
+            the image.
```
Comment (ZwwWayne) [inline mmdet/core/bbox/coder/tblr_bbox_coder.py:21]: Default: True.

File: mmdet/core/bbox/coder/tblr_bbox_coder.py [hunk 8]
```diff
@@ -136,6 +141,8 @@ def tblr2bboxes(priors,
           normalized by the side length (wh) of prior bboxes.
         max_shape (tuple, optional): Shape of the image. Decoded bboxes
           exceeding which will be clamped.
+        clip_border (bool): Whether clip the objects outside the border of
+            the image.
```
Comment (ZwwWayne) [inline mmdet/core/bbox/coder/tblr_bbox_coder.py:145]: Defaults to True.

**Review by ZwwWayne — APPROVED** [review]

**Comment by gritYCDA** [comment]
YOLOBBoxCoder doesn't have unclip_border, may I know why?

**Pull request merged**